lib/bridge/IrcHandler.js

```diff
@@ -742,6 +746,30 @@ IrcHandler.prototype.onMode = Promise.coroutine(function*(req, server, channel,
     yield Promise.all(promises);
 });
 
+IrcHandler.prototype._onModeratedChannelToggle = Promise.coroutine(function*(req, server, channel,
+                                                by, enabled, arg) {
+    let matrixRooms = yield this.ircBridge.getStore().getMatrixRoomsForChannel(server, channel);
+    // modify power levels for all mapped rooms to set events_default to something >0 so by default
+    // people CANNOT speak into it (unless they are a mod or have voice, both of which need to be
+    // configured correctly in the config file).
+    const botClient = this.ircBridge.getAppServiceBridge().getIntent().getClient();
+    for (let i = 0; i < matrixRooms.length; i++) {
+        const roomId = matrixRooms[i].getId();
+        try {
+            const plContent = yield botClient.getStateEvent(roomId, "m.room.power_levels", "");
+            plContent.events_default = enabled ? 1 : 0;
```

Comment:
No, Synapse *should* make sure that every room has a `m.room.power_levels` event, and you can't delete state. The CS API states it'll 404 if the event doesn't exist which will throw instead of returning `undefined`.